docs/builds/guides/migration/migration-to-29.md

```diff
@@ -39,77 +40,78 @@ Since the appearance of the image in the document depends on the image type (blo
-	...
-	image: {
-		styles: [ {
-			name: 'alignLeft',
-			title: 'Left aligned image',
-			icon: objectLeft,
-			className: 'image-style-align-left'
-		} ]
-	}
-} );
-
-// New code
-Editor.create( document.querySelector( '#editor' ), {
-	...
-	image: {
-		styles: {
-			options: [ {
+	```js
+	// Before v29.0.0
+	Editor.create( document.querySelector( '#editor' ), {
+		...
+		image: {
+			styles: [ {
 				name: 'alignLeft',
 				title: 'Left aligned image',
 				icon: objectLeft,
-				// Image types (names of the model elements) supporting this style.
-				modelElements: [ 'imageBlock', 'imageInline' ],
 				className: 'image-style-align-left'
 			} ]
 		}
-	}
-} );
-```
-* Also, several changes has been made to the {@link module:image/imagestyle~ImageStyle} plugin API:
-	* In the {@link module:image/imagestyle/utils image style utils} module
-		* the `defaultIcons` are renamed to {@link module:image/imagestyle/utils~DEFAULT_ICONS},
-		* the `defaultStyles` are renamed to {@link module:image/imagestyle/utils~DEFAULT_OPTIONS},
-		* the `normalizeImageStyles` function is now protected and is renamed to {@link module:image/imagestyle/utils~normalizeStyles}
-	* The `ImageStyleCommand#defaultStyle` and `ImageStyleCommand#styles` are private properties now.
+	} );
+
+	// Since v29.0.0
+	Editor.create( document.querySelector( '#editor' ), {
+		...
+		image: {
+			styles: {
+				options: [ {
+					name: 'alignLeft',
+					title: 'Left aligned image',
+					icon: objectLeft,
+					// Image types (names of the model elements) supporting this style.
+					modelElements: [ 'imageBlock', 'imageInline' ],
+					className: 'image-style-align-left'
+				} ]
+			}
+		}
+	} );
+	```
+
+* Several changes has been also made to the {@link module:image/imagestyle~ImageStyle} plugin API:
+	* In the {@link module:image/imagestyle/utils image style utils} module:
+		* the `defaultIcons` were renamed to {@link module:image/imagestyle/utils~DEFAULT_ICONS},
+		* the `defaultStyles` were renamed to {@link module:image/imagestyle/utils~DEFAULT_OPTIONS},
+		* the `normalizeImageStyles()` function was removed from the public API.
+	* The `ImageStyleCommand#defaultStyle` and `ImageStyleCommand#styles` were removed from the public API.
 
 ## Image toolbar
 
-Due to the changes mentioned above, the {@link module:image/image~ImageConfig#toolbar image toolbar} became crucial in terms of providing the user with proper interaction with images in terms of managing the image type and caption. Thus, it is recommended to use one of the following configurations as the minimum set-up for the image toolbar:
+Due to the changes mentioned above, the {@link module:image/image~ImageConfig#toolbar image toolbar} became crucial in terms of providing tools for proper interaction with images in terms of managing the image type and caption. Thus, it is recommended to use one of the following configurations as the minimum set-up for the image toolbar:
```

Comment:
> is unclear to me. Do I need to configure this in v29.0.0+ or the feature will not work? It's more an announcement than migration to me. I think we need to think this over.

Well, that's complicated 🤦 . It's going to work - it is still possible to insert an inline image in the middle of the paragraph and a block image in the empty paragraph. It is even possible to change the image type by dragging, but only if it has no caption. 

The caption won't work without the toolbar, but this one is handled in the caption section. Maybe the rest of the information should be placed in the image styles section. But IMO it's still about the migration. The other day it was completely fine to have images in the editor without the `ImageStyle` and `ImageToolbar` plugins, but since the inline images, it just gets handicapped, although it works in the basics. As an integrator, where should I get information about that? I'm pretty sure I wouldn't read the whole images guide, especially if I'm just upgrading the version.

Reply:
We try incorporating this explanation into the guide. Even if as something short, like "The image feature will still work if no changes are made to the contextual toolbar configuration. Some of the functions, however, will not work as expected (like captions) and some will only available with limited interaction (e.g. by dragging the image around, while there is a ready toolbar item solution).  
  
How 'bout it?

Reply:
On second thought, we may also add a warning info-box in the guides but keep this here as well.